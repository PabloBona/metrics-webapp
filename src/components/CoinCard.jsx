import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const CoinCard = ({ coin }) => (
  <div className="coin-card col-6 col-sm-4 col-md-4 col-lg-3">
    <Link to={`/coins/${coin.id}`} className="card-link">
      <div className="card border-0 rounded-0 m-0">
        <div className="align-items-sm-center card-body d-flex flex-column">
          <div className="align-self-center">
            <img className="iconStyle p-1 rounded-5" src={coin.icon} alt="" />
          </div>
          <div className="">
            <h3 className="fs-4 fw-bold my-3 text-center text-white">{coin.name}</h3>
          </div>
          <div className="align-self-center">
            <p className="fs-5 text-center text-white m-0">
              {`$${coin.price.toFixed(5)}`}
            </p>
          </div>
        </div>
      </div>
    </Link>
  </div>
);

CoinCard.propTypes = {
  coin: PropTypes.shape({
    id: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
  }).isRequired,
};

export default CoinCard;
