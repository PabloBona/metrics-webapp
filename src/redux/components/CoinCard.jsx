/* eslint-disable react/prop-types */
import "../style/home.css";

const CoinCard = ({ coin }) => {
  
  return (
    <div className="col-6 col-md-3 col-lg-3">
      <div className="card border-0">
        <div className="card-body odd">
          <img className="iconStyle" src={coin.icon} alt="" />
          <h2>{coin.name}</h2>
          <p>Price: {coin.price.toFixed(2)}</p>
        </div>
      </div>
    </div>
  );
};

export default CoinCard;
