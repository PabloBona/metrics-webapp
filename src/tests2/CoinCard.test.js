import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import CoinCard from '../components/CoinCard';

jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
  useSelector: jest.fn(),
}));

const mockData = {
  id: 'bitcoin',
  icon: 'https://static.coinstats.app/coins/1650455588819.png',
  name: 'Bitcoin',
  price: 6362.74960614,
  rank: 1,
};

const MockedCoinCard = ({ coin }) => (
  <BrowserRouter>
    <CoinCard coin={coin} />
  </BrowserRouter>
);

MockedCoinCard.propTypes = {
  coin: PropTypes.shape({
    id: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    rank: PropTypes.number.isRequired,
  }).isRequired,
};

describe('CoinCard Component', () => {
  test('renders coin name', () => {
    render(<MockedCoinCard coin={mockData} />);
    const coinNameElement = screen.getByText(mockData.name);
    expect(coinNameElement).toBeInTheDocument();
  });

  test('renders coin price', () => {
    render(<MockedCoinCard coin={mockData} />);
    const coinPriceElement = screen.getByText(`$${mockData.price.toFixed(5)}`);
    expect(coinPriceElement).toBeInTheDocument();
  });

  test('renders coin rank', () => {
    render(<MockedCoinCard coin={mockData} />);
    const coinRankElement = screen.getByText(`Rank ${mockData.rank}`);
    expect(coinRankElement).toBeInTheDocument();
  });

  test('renders link with correct URL', () => {
    render(<MockedCoinCard coin={mockData} />);
    const linkElement = screen.getByRole('link');
    expect(linkElement).toHaveAttribute('href', `/coins/${mockData.id}`);
  });

  // Add more test cases as needed for other parts of the CoinCard component
});
