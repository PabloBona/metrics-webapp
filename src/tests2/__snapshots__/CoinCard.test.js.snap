// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoinCard Component renders coin name 1`] = `
<h3
  class="coin-name  my-3 text-center text-white"
>
  Bitcoin
</h3>
`;

exports[`CoinCard Component renders coin price 1`] = `
<p
  class="fs-5 text-center text-white m-0"
>
  $6362.74961
</p>
`;

exports[`CoinCard Component renders coin rank 1`] = `
<p
  class="fs-5 text-center text-white m-0"
>
  Rank 1
</p>
`;

exports[`CoinCard Component renders link with correct URL 1`] = `
<a
  class="card-link"
  href="/coins/bitcoin"
>
  <div
    class="card border-0 rounded-0 m-0"
  >
    <div
      class="align-items-sm-center card-body d-flex flex-column"
    >
      <div
        class="align-self-center"
      >
        <img
          alt=""
          class="iconStyle p-1 rounded-5"
          src="https://static.coinstats.app/coins/1650455588819.png"
        />
      </div>
      <div>
        <h3
          class="coin-name  my-3 text-center text-white"
        >
          Bitcoin
        </h3>
      </div>
      <div
        class="align-self-center container-details"
      >
        <p
          class="fs-5 text-center text-white m-0"
        >
          $6362.74961
        </p>
        <p
          class="fs-5 text-center text-white m-0"
        >
          Rank 1
        </p>
      </div>
    </div>
  </div>
</a>
`;
